spec/controllers/contacts_controller_spec.rb

```diff
@@ -0,0 +1,159 @@
+  let(:valid_attributes) {
+    skip("Add a hash of attributes valid for your model")
+  }
+
+  let(:invalid_attributes) {
+    skip("Add a hash of attributes invalid for your model")
+  }
+
+  # This should return the minimal set of values that should be in the session
+  # in order to pass any filters (e.g. authentication) defined in
+  # ContactsController. Be sure to keep this updated too.
+  let(:valid_session) { {} }
+
+  describe "GET index" do
+    it "assigns all contacts as @contacts" do
+      contact = Contact.create! valid_attributes
+      get :index, {}, valid_session
+      expect(assigns(:contacts)).to eq([contact])
+    end
+  end
+
+  describe "GET show" do
+    it "assigns the requested contact as @contact" do
+      contact = Contact.create! valid_attributes
+      get :show, {:id => contact.to_param}, valid_session
+      expect(assigns(:contact)).to eq(contact)
+    end
+  end
+
+  describe "GET new" do
+    it "assigns a new contact as @contact" do
+      get :new, {}, valid_session
+      expect(assigns(:contact)).to be_a_new(Contact)
+    end
+  end
+
+  describe "GET edit" do
+    it "assigns the requested contact as @contact" do
+      contact = Contact.create! valid_attributes
+      get :edit, {:id => contact.to_param}, valid_session
+      expect(assigns(:contact)).to eq(contact)
+    end
+  end
+
+  describe "POST create" do
+    describe "with valid params" do
+      it "creates a new Contact" do
+        expect {
+          post :create, {:contact => valid_attributes}, valid_session
+        }.to change(Contact, :count).by(1)
+      end
+
+      it "assigns a newly created contact as @contact" do
+        post :create, {:contact => valid_attributes}, valid_session
+        expect(assigns(:contact)).to be_a(Contact)
+        expect(assigns(:contact)).to be_persisted
+      end
+
+      it "redirects to the created contact" do
+        post :create, {:contact => valid_attributes}, valid_session
+        expect(response).to redirect_to(Contact.last)
+      end
+    end
+
+    describe "with invalid params" do
+      it "assigns a newly created but unsaved contact as @contact" do
+        post :create, {:contact => invalid_attributes}, valid_session
+        expect(assigns(:contact)).to be_a_new(Contact)
+      end
+
+      it "re-renders the 'new' template" do
+        post :create, {:contact => invalid_attributes}, valid_session
+        expect(response).to render_template("new")
+      end
+    end
+  end
+
+  describe "PUT update" do
+    describe "with valid params" do
+      let(:new_attributes) {
+        skip("Add a hash of attributes valid for your model")
+      }
+
+      it "updates the requested contact" do
+        contact = Contact.create! valid_attributes
+        put :update, {:id => contact.to_param, :contact => new_attributes}, valid_session
+        contact.reload
+        skip("Add assertions for updated state")
+      end
+
+      it "assigns the requested contact as @contact" do
```

Comment:
Prefer single-quoted strings when you don't need string interpolation or special symbols.